### PR TITLE
Check if switch has machine connections before deleting or force delete

### PR DESCRIPTION
**Pull request by iljarotar**

### CODE REVIEW

**Review by majst01 — APPROVED** [review]

File: cmd/metal-api/internal/service/switch-service.go [hunk 1]
```diff
@@ -80,9 +81,11 @@ func (r *switchResource) webService() *restful.WebService {
 		Operation("deleteSwitch").
 		Doc("deletes an switch and returns the deleted entity").
 		Param(ws.PathParameter("id", "identifier of the switch").DataType("string")).
+		Param(ws.PathParameter("force", "if true switch is deleted with no validation").DataType("boolean").DefaultValue("false")).
 		Metadata(restfulspec.KeyOpenAPITags, tags).
 		Writes(v1.SwitchResponse{}).
 		Returns(http.StatusOK, "OK", v1.SwitchResponse{}).
+		Returns(http.StatusBadRequest, "Bad input data", httperrors.HTTPErrorResponse{}).
```
Comment (Gerrit91) [inline cmd/metal-api/internal/service/switch-service.go:88]: FYI: We typically do not include this response type explicitly in the API as it goes for every request. We handle this through the default error.

File: cmd/metal-api/internal/service/switch-service.go [hunk 2]
```diff
@@ -191,13 +194,22 @@ func (r *switchResource) findSwitches(request *restful.Request, response *restfu
 
 func (r *switchResource) deleteSwitch(request *restful.Request, response *restful.Response) {
 	id := request.PathParameter("id")
+	force, err := strconv.ParseBool(request.PathParameter("force"))
+	if err != nil {
+		r.sendError(request, response, httperrors.BadRequest(err))
```
Comment (Gerrit91) [inline cmd/metal-api/internal/service/switch-service.go:199]: Missing return. Code will continue after this.

Comment (majst01) [reply cmd/metal-api/internal/service/switch-service.go:199]: Shit, sorry

Comment (majst01) [reply cmd/metal-api/internal/service/switch-service.go:199]: @iljarotar please create a FIX PR

File: cmd/metal-api/internal/service/switch-service.go [hunk 3]
```diff
@@ -80,9 +81,11 @@ func (r *switchResource) webService() *restful.WebService {
 		Operation("deleteSwitch").
 		Doc("deletes an switch and returns the deleted entity").
 		Param(ws.PathParameter("id", "identifier of the switch").DataType("string")).
+		Param(ws.PathParameter("force", "if true switch is deleted with no validation").DataType("boolean").DefaultValue("false")).
```
Comment (Gerrit91) [inline cmd/metal-api/internal/service/switch-service.go:84]: Also this needs to be a QueryParam. This leads to a broken swagger spec?

**Pull request merged**